go-jsref.Resolver.Resolve() panics when enum has null-value

My schema has such field
```
"status": {
                                "type": ["string", "null"],
                                "enum": [
                                    "sent",
                                    "duplicate",
                                    "error",
                                    "invalid",
                                    "rejected",
                                    "unqueued",
                                    "unsubscribed",
                                    null
                                ]
                            }
```

and go-jsref.Resolver.Resolve() panics

```
panic: reflect: call of reflect.Value.Interface on zero Value

goroutine 1 [running]:
panic(0x61c800, 0xc4201333a0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
reflect.valueInterface(0x0, 0x0, 0x0, 0x1, 0x0, 0x0)
        /usr/local/go/src/reflect/value.go:912 +0x204
reflect.Value.Interface(0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/reflect/value.go:907 +0x44
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc420138f80, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:163 +0x205
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc420138f20, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc420138830, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc4201387d0, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc420138780, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc420138700, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc4201386a0, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x619640, 0xc420138510, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.traverseExpandRefRecursive(0xc420127cb0, 0xc420127f08, 0x61e960, 0xc4201111a0, 0x15, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:183 +0x606
gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref.(*Resolver).Resolve(0xc420127f08, 0x61e960, 0xc4201111a0, 0x0, 0x0, 0xc420127e08, 0x1, 0x1, 0x0, 0x0, ...)
        /home/d.telyukh/go/src/gitlab.2gis.ru/mailer/mailer-docs/go-jsschema-builder/vendor/github.com/lestrrat/go-jsref/jsref.go:111 +0x37f
main.processFiles(0x662e54, 0xa, 0xc420127f08)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,5 +1,5 @@
 language: go
 sudo: false
 go:
-    - 1.6
+    - 1.7.x
     - tip

--- a/jsref.go
+++ b/jsref.go
@@ -155,6 +155,12 @@ func traverseExpandRefRecursive(ctx *resolveCtx, r *Resolver, rv reflect.Value) 
 				elem = elem.Elem()
 			}
 
+			// Need to check for elem being Valid, otherwise the
+			// subsequent call to Interface() will fail
+			if !elem.IsValid() {
+				continue
+			}
+
 			if elemcontainer.IsValid() {
 				if !elemcontainer.CanSet() {
 					continue

--- a/jsref_test.go
+++ b/jsref_test.go
@@ -265,3 +265,31 @@ func TestHyperSchemaRecursive(t *testing.T) {
 		}
 	}
 }
+
+func TestGHIssue7(t *testing.T) {
+	src := []byte(`{
+  "status": {
+    "type": ["string", "null"],
+    "enum": [
+      "sent",
+      "duplicate",
+      "error",
+      "invalid",
+      "rejected",
+      "unqueued",
+      "unsubscribed",
+      null
+    ]
+  }
+}`)
+
+	var v interface{}
+	if !assert.NoError(t, json.Unmarshal(src, &v), `Unmarshal should succeed`) {
+		return
+	}
+
+	res := jsref.New()
+	result, err := res.Resolve(v, "", jsref.WithRecursiveResolution(true))
+	t.Logf("%s", result)
+	t.Logf("%s", err)
+}

